cpp/cmake/thirdparty/CUDF_GetThrust.cmake

```diff
@@ -0,0 +1,33 @@
+#=============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_thrust VERSION)
+    CPMFindPackage(NAME Thrust
```

Comment:
Do we actually want to find thrust in this situation given we have a patch that we want to apply? Or do we always want to fetch it?

Reply:
You're right, we should always add our own thrust to `_deps`. I'll need to check if there's a way to disregard the `CPM_SOURCE_CACHE` for this too.